lib/client.ml

```diff
@@ -1,26 +1,27 @@
 open Lwt_unix
 
 type 'a t = {
-  address : string;
-  port : int;
+  address : Address.t;
   socket : file_descr;
   state : 'a ref;
   recv_mutex : Lwt_mutex.t;
   state_mutex : Lwt_mutex.t;
 }
 
-let peer_from (client : 'a t) =
-  Peer.
-    {
-      address = client.address;
-      port = client.port;
-      known_peers = [];
-      state = Alive;
-    }
+let address_of { address; _ } = address
+
+let peer_from { address; _ } =
```

Comment:
`to_peer` might be better, since we are turning a client into a peer. `Client.to_peer client_a` returns a peer with the address of `client_a`, I think that makes sense.